presto-parser/src/main/java/com/facebook/presto/sql/ExpressionFormatter.java

```diff
@@ -566,6 +570,43 @@ static String formatSortItems(List<SortItem> sortItems, boolean unmangleNames)
                 .iterator());
     }
 
+    static String formatGroupBy(List<GroupingElement> groupingElements)
+    {
+        checkState(groupingElements.stream()
+                .filter(groupingElement -> !(groupingElement instanceof SimpleGroupBy))
+                .count() <= 1, "multiple groupings are not yet supported");
+
+        List<Expression> simpleGroupings = groupingElements.stream()
+                .filter(groupingElement -> groupingElement instanceof SimpleGroupBy)
```

Comment:
If you define the formatting of SimpleGroupBy to be:

`x` if SimpleGroupBy contains a single expression
`(x1, x2, ...)` if it contains multiple expressions

it should all just work out.

For instance:

```
GROUP BY a, (b, c)
```
would parse as 
```
- GroupBy
    - SimpleGroupBy: [a]
    - SimpleGroupBy: [b, c]
```
And formatting it would just reverse that.